### PR TITLE
BSVE integration plugin

### PR DESCRIPTION
**Pull request by jbeezley**
This adds the server side component that validates an authentication token through the BSVE API, and if valid returns a girder authentication.  A new girder user is created from the BSVE account information if necessary.  The API is nearly identical to girder's own `GET /user/authenticate`, but takes an email/token pair instead of user/password.

Before deploying this we should ensure that registration is closed through the girder api because it would introduce a security hole.

### CODE REVIEW

File: bsve/server/auth.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        resp = resp.json()
+        if resp['status'] == 1:
+            return resp['data']
+        return None
+
+    def _generateLogin(self, email):
+        """Generate a login name from an email address."""
+        user = email.lower()
+
+        # poor man's escaping, probably not terribly secure
```
Comment (mgrauer) [inline bsve/server/auth.py]: what is the danger here?  or what did you mean by secure?

Comment (jbeezley) [reply bsve/server/auth.py]: A login name collision mostly.  We are bypassing some of the safety valves that girder provides in the default registration/authentication endpoints.  By closing registrations by normal means, this should be good enough.

Comment (jbeezley) [reply bsve/server/auth.py]: I can remove the comment if it's too scary.

**Comment by mgrauer** [comment]
This looks good to me, in that I've read through all the code and it seems correct.

Let me know if you need me to build it on the beta instance, or if @dorukozturk can do that to test it live.

File: bsve/server/auth.py (hunk 1 above)
Comment (mgrauer) [reply bsve/server/auth.py]: Can you adjust the comment to mention what you said here?  This risk seems reasonable and low, better than less specific comment about security.

**Comment by jbeezley** [comment]
I updated the client code to perform the authentication and start up a default session.  It works on dev.bsvecosystem.net and qa.bsvecosystem.net, but the [api url](https://github.com/OpenGeoscience/minerva-docker/pull/6/files#diff-510279c0b4b2d3511a30a31bc33e097aR196) will have to be updated for the production site.

**Comment by aashish24** [comment]
@jbeezley the last commit looks great. Can we merge this branch into master or are there any major concerns?

**Comment by jbeezley** [comment]
Yes, this should probably be merged.

**Pull request merged**